CoreScriptsRoot/Modules/Settings/Pages/Players.lua

```diff
@@ -117,12 +134,42 @@ local function Initialize()
 					if friendLabel and friendLabelText and friendLabelText.Text ~= '' then
 						friendLabel.ImageTransparency = 1
 						friendLabelText.Text = ''
-						if localPlayer and player then
+					end
+					if AtFriendLimit(localPlayer) then
+						while PromptCreator:IsCurrentlyPrompting() do
+							wait()
+						end
+						PromptCreator:CreatePrompt({
+							WindowTitle = "Friend Limit Reached",
+							MainText = string.format("You can not send a friend request to %s because you are at the max friend limit.", player.Name),
+							ConfirmationText = "Okay",
+							CancelActive = false,
+							Image = BUST_THUMBNAIL_URL ..player.UserId,
+							ImageConsoleVR = THUMBNAIL_URL ..player.UserId,
+							StripeColor = Color3.fromRGB(183, 34, 54),
+						})
+						friendStatusCreate(playerLabel, player)
+					else
+						if AtFriendLimit(player) then
+							while PromptCreator:IsCurrentlyPrompting() do
+								wait()
+							end
+							PromptCreator:CreatePrompt({
+								WindowTitle = "Error Sending Friend Request",
+								MainText = string.format("You can not send a friend request to %s because they are at the max friend limit.",  player.Name),
+								ConfirmationText = "Okay",
+								CancelActive = false,
+								Image = BUST_THUMBNAIL_URL ..player.UserId,
+								ImageConsoleVR = THUMBNAIL_URL ..player.UserId,
+								StripeColor = Color3.fromRGB(183, 34, 54),
+							})
+							friendStatusCreate(playerLabel, player)
+						else
 							localPlayer:RequestFriendship(player)
 						end
 					end
 				end
-				local friendLabel2, friendLabelText2 = utility:MakeStyledButton("FriendStatus", "Add Friend", UDim2.new(0, 182, 0, 46), addFriendFunc)
+				local friendLabel2, friendLabelText2 = utility:MakeStyledButton("FriendStatus", status == Enum.FriendStatus.FriendRequestReceived and "Accept Request" or "Add Friend", UDim2.new(0, 182, 0, 46), addFriendFunc)
```

Comment:
Is there any reason why this can't just assign directly to friendLabel and friendLabelText? I know this wasn't your change but this could be changed for a bit of added simplicity.